### PR TITLE
Add github actions

### PR DESCRIPTION
**Pull request by elezar**
This PR does the following:
* Update the golang version to v1.20.4 to allow for the use of the k8s-test-infra images
* Use CUDA base images to minimize the risk of GLIBC version mismatches caused by Golang updates
* Switch to using the k8s-test-infra images
* Add GitHub CI actions

### CODE REVIEW

File: .github/workflows/image.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build image
+        env:
+          IMAGE_NAME: ghcr.io/${LOWERCASE_REPO_OWNER}/k8s-device-plugin
+          VERSION: ${COMMIT_SHORT_SHA}
+          # TODO: We could also only set this to true for pushed images.
+          BUILD_MULTI_ARCH_IMAGES: "true"
+        run: |
+          echo "${VERSION}"
+          make -f deployments/container/Makefile build-${{ matrix.dist }}
```
Comment (ArangoGutierrez) [inline .github/workflows/image.yaml:76]:
```suggestion
          make -f deployments/gpu-operator/Makefile build-${{ matrix.dist }}
```

Edit here, or rename the path to the Dockerfile

File: .github/workflows/image.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build image
+        env:
+          IMAGE_NAME: ghcr.io/${LOWERCASE_REPO_OWNER}/k8s-device-plugin
+          VERSION: ${COMMIT_SHORT_SHA}
+          # TODO: We could also only set this to true for pushed images.
+          BUILD_MULTI_ARCH_IMAGES: "true"
```
Comment (ArangoGutierrez) [inline .github/workflows/image.yaml:73]: Let's move this to the `Calculate build vars` section so is only triggered at merge

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]
Thanks Evan

File: .github/workflows/image.yaml (hunk 1 above)
Comment (elezar) [reply .github/workflows/image.yaml]: I have edited here and created #33 for follow up.

File: .github/workflows/image.yaml (hunk 2 above)
Comment (elezar) [reply .github/workflows/image.yaml]: Done.

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]
Let's remove all Blossom related actions, there's no need for them

`Run pre sanity / build (pull_request)`

File: .github/workflows/image.yaml (hunk 2 above)
Comment (elezar) [reply .github/workflows/image.yaml]: Done.

**Review by ArangoGutierrez — APPROVED** [review]

File: .github/workflows/golang.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
+# Copyright 2023 NVIDIA CORPORATION
```
Comment (ArangoGutierrez) [inline .github/workflows/golang.yaml:1]:
```suggestion
# Copyright 2024 NVIDIA CORPORATION
```

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]
One last nit

File: .github/workflows/golang.yaml (hunk 3 above)
Comment (elezar) [reply .github/workflows/golang.yaml]: Done. This was copied from your PR.

Comment (ArangoGutierrez) [reply .github/workflows/golang.yaml]: I have fixed my PR, thanks for noticing

File: docker/Dockerfile.devel [hunk 4]
```diff
@@ -11,7 +11,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-ARG GOLANG_VERSION=1.20.1
+ARG GOLANG_VERSION=1.20.4
```
Comment (ArangoGutierrez) [inline docker/Dockerfile.devel:14]: Shouldn't we rm this file now?

Comment (elezar) [reply docker/Dockerfile.devel:14]: Hmmm. I thought I did.

Comment (elezar) [reply docker/Dockerfile.devel]: Done.

File: deployments/gpu-operator/Dockerfile.ubi8 [hunk 5]
```diff
@@ -12,18 +12,33 @@
...
-# image based on the target OS to build the golang executables here -- especially
-# if cgo code is included.
-FROM golang:${GOLANG_VERSION} AS build
+RUN yum install -y \
+    wget make git gcc \
+     && \
+    rm -rf /var/cache/yum/*
 
-SHELL ["/bin/bash", "-c"]
+ARG GOLANG_VERSION=x.x.x
```
Comment (ArangoGutierrez) [inline deployments/gpu-operator/Dockerfile.ubi8:26]: ARG defined twice, no need to be defined before `FROM` only here

File: deployments/gpu-operator/Dockerfile.ubuntu [hunk 6]
```diff
@@ -12,18 +12,33 @@
...
-# image based on the target OS to build the golang executables here -- especially
-# if cgo code is included.
-FROM golang:${GOLANG_VERSION} AS build
+RUN apt-get update && \
+    apt-get install -y wget make git gcc curl \
+    && \
+    rm -rf /var/lib/apt/lists/*
 
-SHELL ["/bin/bash", "-c"]
+ARG GOLANG_VERSION=x.x.x
```
Comment (ArangoGutierrez) [inline deployments/gpu-operator/Dockerfile.ubuntu:26]: same comment

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]

File: deployments/gpu-operator/Dockerfile.ubuntu (hunk 6 above)
Comment (elezar) [reply deployments/gpu-operator/Dockerfile.ubuntu]: OK. I've updated this.

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm

**Pull request merged**